### PR TITLE
fix function name check

### PR DESCRIPTION
**Pull request by GiggleLiu**
Skipped a check that gives error for non-symbolic function names.
Not sure this fix is proper, but it works for my application.

### CODE REVIEW

**Comment by vchuravy** [comment]
bors rty

**Comment by vchuravy** [comment]
bors try

**Comment by vchuravy** [comment]
Can you add a test that makes sure that one can overload such kernels?

**Comment by GiggleLiu** [comment]
> Can you add a test that makes sure that one can overload such kernels?

Yes, there is such a test.

File: test/test.jl [hunk 1]
```diff
@@ -216,3 +216,15 @@ if has_cuda_gpu()
     @test MultiEvent((event1, event2, event3)) isa Event
   end
 end
+
+  @testset "fallback test: callable types" begin
+    function f end
+    @kernel function (a::typeof(f))(x, ::Val{m}) where m
```
Comment (vchuravy) [inline test/test.jl]: Can you add a second test:

```
     @kernel function (a::typeof(f))(x, ::Val{1})
```

to see what happens when two invocations of `@kernel` define the wrappers? You should get a warning from Julia

Comment (GiggleLiu) [reply test/test.jl:253]: I added a test, everything so far works fine? But I didn't get a warning.

**Comment by GiggleLiu** [comment]
Hi, is it good to merge, or are there some further suggestions?

**Pull request merged**